statistics/handle/handle.go

```diff
@@ -413,11 +416,59 @@ func (h *Handle) MergePartitionStats2GlobalStatsByTableID(sc sessionctx.Context,
 		return
 	}
 	globalTableInfo := globalTable.Meta()
-	return h.mergePartitionStats2GlobalStats(sc, opts, is, globalTableInfo, isIndex, histIDs)
+	return h.mergePartitionStats2GlobalStats(sc, opts, is, globalTableInfo, isIndex, histIDs, globalPartitionStats)
+}
+
+// GetTableAllPartitionStats return all partition stats of a table
+func (h *Handle) GetTableAllPartitionStats(is infoschema.InfoSchema, physicalID int64) (map[int64]*statistics.Table, error) {
+	h.mu.Lock()
+	globalTable, ok := h.getTableByPhysicalID(is, physicalID)
+	h.mu.Unlock()
+	if !ok {
+		return nil, errors.Errorf("unknown physical ID %d in stats meta table, maybe it has been dropped", physicalID)
+	}
+	globalTableInfo := globalTable.Meta()
```

Comment:
Maybe there exists some way to get `TableInfo` faster than `get statistics.Table, then get TableInfo`.

Reply:
removed.